krod/core/llm_manager.py

```
@@ -182,42 +182,98 @@ async def _call_openai_chat(
                 "success": True,
-                "text": choices[0]["message"]["content"],
+                "text": content,
                 "usage": response.get("usage", {}),
-                "model": response.get("model", self.config.get("default_model", "unknown")),
-                "prompt_type": "text"
+                "model": response.get("model", model),
+                "prompt_type": "text",
+                "finish_reason": choices[0].get("finish_reason")
             }
 
+        except openai.RateLimitError as e:
```

Ruff (0.11.9)

Undefined name `openai`

(F821)